akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/auction/AuctionEntity.java

```diff
@@ -0,0 +1,241 @@
+    } else if (bid.getBidPrice() <= currentBidMaximum) {
+      return handleAutomaticOutbid(bid, auction, now, currentBid, currentBidPrice, currentBidMaximum);
+    } else {
+      return handleNewWinningBidder(bid, auction, now, currentBidMaximum);
+    }
+  }
+
+  /**
+   * Handle the situation where a bid will be accepted, but it will be automatically outbid by the current bidder.
+   *
+   * This emits two events, one for the bid currently being replace, and another automatic bid for the current bidder.
+   */
+  private Effect<AuctionEvent, AuctionState> handleAutomaticOutbid(
+      PlaceBid bid, Auction auction, Instant now, Optional<Bid> currentBid, int currentBidPrice, int currentBidMaximum) {
+    // Adjust the bid so that the increment for the current maximum makes the current maximum a valid bid
+    int adjustedBidPrice = Math.min(bid.getBidPrice(), currentBidMaximum - auction.getIncrement());
+    int newBidPrice = adjustedBidPrice + auction.getIncrement();
+
+    return Effect().persist(Arrays.asList(
+        new BidPlaced(entityUUID,
+            new Bid(bid.getBidder(), now, adjustedBidPrice, bid.getBidPrice())
+        ),
+        new BidPlaced(entityUUID,
+            new Bid(currentBid.get().getBidder(), now, newBidPrice, currentBidMaximum)
+        )
+      ))
+      .thenReply(bid, newState -> new PlaceBidResult(PlaceBidStatus.ACCEPTED_OUTBID, newBidPrice, currentBid.get().getBidder()));
+  }
+
+  /**
+   * Handle the situation where a bid will be accepted as the new winning bidder.
+   */
+  private Effect<AuctionEvent, AuctionState> handleNewWinningBidder(PlaceBid bid,
+                                                       Auction auction, Instant now, int currentBidMaximum) {
+    int nextIncrement = Math.min(currentBidMaximum + auction.getIncrement(), bid.getBidPrice());
+    int newBidPrice;
+    if (nextIncrement < auction.getReservePrice()) {
+      newBidPrice = Math.min(auction.getReservePrice(), bid.getBidPrice());
+    } else {
+      newBidPrice = nextIncrement;
+    }
+    return Effect().persist(new BidPlaced(
+        entityUUID,
+        new Bid(bid.getBidder(), now, newBidPrice, bid.getBidPrice())
+      ))
+      .thenReply(bid, newState -> {
+        PlaceBidStatus status;
+        if (newBidPrice < auction.getReservePrice()) {
+          status = PlaceBidStatus.ACCEPTED_BELOW_RESERVE;
+        } else {
+          status = PlaceBidStatus.ACCEPTED;
+        }
+        return new PlaceBidResult(status, newBidPrice, bid.getBidder());
+      });
+  }
+
+  @Override
+  public AuctionState emptyState() {
+    return AuctionState.notStarted();
+  }
+
+  @Override
+  public CommandHandler<AuctionCommand, AuctionEvent, AuctionState> commandHandler() {
+    return notStartedHandler
+        .orElse(underAuctionHandler)
+        .orElse(completedHandler)
+        .orElse(cancelledHandler)
+        .orElse(getAuctionHandler)
+        .orElse(cancelledHandler)
```

Comment:
nitpicking: 
`cancelledHandler` is used twice, instead of `canceldHandler` and `cancelledHandler`